3party/CMakeLists.txt

```diff
@@ -41,6 +41,10 @@ if (NOT WITH_SYSTEM_PROVIDED_3PARTY)
 
   # Add protobuf library.
   add_subdirectory(protobuf)
+
+  add_library(utf8cpp INTERFACE)
+  add_library(utf8cpp::utf8cpp ALIAS utf8cpp)
+  target_include_directories(utf8cpp INTERFACE "${OMIM_ROOT}/3party/utfcpp/source")
```

Comment:
Why this change was required? Can it be extracted into a separate PR to test and polish properly?

Reply:
1) Remove hardcoded `"${OMIM_ROOT}/3party/utf8cpp/include"` from include path
2) Unify usage for system-provided and vendored copy of utfcpp.

Changes in this PR somewhat depend on each other, I really wouldn't like to split them.